cpg-library/src/main/java/de/fraunhofer/aisec/cpg/analysis/ValueEvaluator.kt

```diff
@@ -71,18 +82,36 @@ class ValueEvaluator(
 
     /** Tries to evaluate this expression. Anything can happen. */
     fun evaluate(expr: Expression?): Any? {
+        // Add the expression to the current path
         expr?.let { this.path += it }
 
         when (expr) {
+            // For a literal, we can just take its value, and we are finished
             is Literal<*> -> {
                 return expr.value
             }
-            is DeclaredReferenceExpression -> return resolveDeclaration(expr.refersTo)
+            // For a reference, we are interested into its last assignment into the reference
+            // denoted by the previous DFG edge
+            is DeclaredReferenceExpression -> {
+                val prevDFG = expr.prevDFG
+
+                // We could potentially have more incoming DFG nodes, so we want to filter for
```

Comment:
Did we not say to abort in this case?

Reply:
We could this as well. At some point we should probably have some kind of error indication, e.g. `TooManySolutions`

Reply:
As per discussion, we abort in this case now.